.github/workflows/patroni_backup.yaml

```diff
@@ -0,0 +1,105 @@
+name: Deploy Patroni(Postgres) backup containers
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment'
+        required: true
+        default: 'dev'
+        type: choice
+        options:
+        - dev
+        - test
+        - prod
+
+jobs:
+  # build the backup patroni container
+  build:
+    name: 'Build Patroni Backup APP'
+    runs-on: ubuntu-latest
+    env:
+      BUILD_ID: ${{ github.event.number }}
+      BUILD_NAMESPACE: d89793-tools
+      NAME: patroni-backup
+      BUILD_TAG: latest
+    steps:
+      # Checkout the PR branch
+      - name: Print env
+        run: |
+          echo BUILD ID: $BUILD_ID
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+      
+      - name: Checkout Target Branch
+        uses: actions/checkout@v1
+      
+      # Log in to OpenShift.
+      # Note: The secrets needed to log in are NOT available if the PR comes from a FORK.
+      # PR's must originate from a branch off the original repo or else all openshift `oc` commands will fail.
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
+
+      # Build backup container
+      - name: Build backup image
+        working-directory: './deployment/openshift/databases'
+        run: |
+          oc -n ${BUILD_NAMESPACE} process -f prebuilt-db-backup-build.yaml \
+          -p NAME=${NAME} \
+          -p TAG=${BUILD_TAG} | oc -n ${BUILD_NAMESPACE} apply -f - --wait
+
+  # deploy the backup image
+  deploy:
+    if: ${{contains(['dev', 'test', 'prod'], github.event.inputs.environment)}}
+    name: Deploy Patroni to ${{github.event.inputs.environment}} environment
+    runs-on: ubuntu-latest
+    env:
+      BUILD_ID: ${{ github.event.number }}
+      BUILD_ENV: ${{ github.event.inputs.environment }}
+      NAMESPACE: d89793-${{ github.event.inputs.environment }}
+      BUILD_NAMESPACE: d89793-tools
+      BRANCH: automated-db-backups
+      NAME: patroni-backup
+      DB_NAME: app
+    needs:
```

Comment:
Probably don't need this `needs` now you only have a single deploy job. You could get rid of the extra checkout action in this case. Also applies to the mongo backup jub

Reply:
Oh and the superfluous extra `oc login` command too. To clarify, I mean just merging the steps from `build` into this job

Reply:
Ohh, `needs` ensure and wait for the build command to be successfully completed. It makes the two jobs sequential. Isn't that the need too?

Also regarding the extra `checkout` and `oc login` you mentioned, my assumption was both these jobs run in their independent containers - so checkout was required. Maybe I am wrong, please let me know if that is not the case and so - these commands are not required?

Reply:
> Isn't that the need too?

Yes but since you now only have a single task in `build` and only one job that depends on it, I would just move the build task into the deploy job, rather than have two separate jobs. Doesn't require a new container to be spun up and checkout/login to be run again. Up to you though, I can see an argument for having build and deploy in isolated jobs in the pipeline.

Reply:
Got it 👍 

Yeah, but I would like to go ahead with different jobs. A couple of reasons:
1. Separation of concerns
2. Both jobs act on a different namespace. While `build` builds to the tools namespace, `deploy` deploys to the respective namespaces for dev, test, and prod. Thereby keeping them separate would logically make more sense.

Please let me know if you think otherwise.